lib/src/layout_element.dart

```diff
@@ -28,52 +31,108 @@ class TableLayoutElement extends LayoutElement {
 
   @override
   Widget toWidget(RenderContext context) {
-    final colWidths = children
-        .where((c) => c.name == "colgroup")
-        .map((group) {
-          return group.children.where((c) => c.name == "col").map((c) {
-            final widthStr = c.attributes["width"] ?? "";
-            if (widthStr.endsWith("%")) {
-              final width =
-                  double.tryParse(widthStr.substring(0, widthStr.length - 1)) *
-                      0.01;
-              return FractionColumnWidth(width);
-            } else {
-              final width = double.tryParse(widthStr);
-              return width != null ? FixedColumnWidth(width) : null;
-            }
-          });
-        })
-        .expand((i) => i)
-        .toList()
-        .asMap();
+    final rows = <TableRowLayoutElement>[];
+    List<TrackSize> columnSizes;
+    for (var child in children) {
+      if (child is TableStyleElement) {
+        // Map <col> tags to predetermined column track sizes
+        columnSizes = child.children.where((c) => c.name == "col").map((c) {
+          final colWidth = c.attributes["width"];
+          if (colWidth != null && colWidth.endsWith("%")) {
+            final percentageSize =
+                double.tryParse(colWidth.substring(0, colWidth.length - 1));
+            return percentageSize != null
+                ? FlexibleTrackSize(percentageSize * 0.01)
+                : FlexibleTrackSize(1);
+          } else if (colWidth != null) {
+            final fixedPxSize = double.tryParse(colWidth);
+            return fixedPxSize != null
+                ? FixedTrackSize(fixedPxSize)
+                : FlexibleTrackSize(1);
+          } else {
+            return FlexibleTrackSize(1);
+          }
+        }).toList(growable: false);
+      } else if (child is TableSectionLayoutElement) {
+        rows.addAll(child.children.whereType());
+      } else if (child is TableRowLayoutElement) {
+        rows.add(child);
+      }
+    }
+
+    // All table rows have a height intrinsic to their (spanned) contents
+    final rowSizes =
+        List.generate(rows.length, (_) => IntrinsicContentTrackSize());
+
+    // Calculate column bounds
+    int columnMax = rows
+        .map((row) => row.children
+            .whereType<TableCellElement>()
+            .fold(0, (int value, child) => value + child.colspan))
+        .fold(0, max);
```

Comment:
This is probably the most creative way I've ever seen to get the `max` from a list. I'm going to use this more often